toolchain/lex/lex.cpp

```diff
@@ -1322,44 +1425,55 @@ class Lexer::ErrorRecoveryBuffer {
   auto InsertBefore(TokenIndex insert_before, TokenKind kind) -> void {
     CARBON_CHECK(insert_before.index > 0)
         << "Cannot insert before the start of file token.";
+    CARBON_CHECK(insert_before.index <
+                 static_cast<int>(buffer_.token_infos_.size()))
+        << "Cannot insert after the end of file token.";
     CARBON_CHECK(new_tokens_.empty() ||
                  new_tokens_.back().first <= insert_before)
         << "Insertions performed out of order.";
 
+    // If the token we're inserting before had leading whitespace, mark the
```

Comment:
Sorry, I'm mentally trying to figure out if "The token we're inserting before" means "inserting previously" (i.e., the token being inserted previously had leading whitespace) or "the next token". Would it work for you to refer directly to the parameter:

```suggestion
    // If the `insert_before` token has leading whitespace, mark the
```

But if you prefer, I think the tense of "has" versus "had" is also load bearing:

```suggestion
    // If the token we're inserting before has leading whitespace, mark the
```